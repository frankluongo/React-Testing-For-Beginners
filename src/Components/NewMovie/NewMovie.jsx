import React from "react";
import MovieForm from "../MovieForm";

const NewMovie = () => {
  return (
    <section>
      <h2 data-testid="page-title">New Movie</h2>
      <MovieForm />
    </section>
  );
};

export default NewMovie;
