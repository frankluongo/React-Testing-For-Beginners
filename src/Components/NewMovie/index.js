import NewMovie from "./NewMovie";

export default NewMovie;
