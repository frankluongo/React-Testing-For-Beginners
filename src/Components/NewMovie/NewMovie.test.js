import React from "react";
import { render, cleanup, fireEvent } from "react-testing-library";

import NewMovie from "./NewMovie";

afterEach(cleanup);

test("<NewMovie />", () => {
  const { getByTestId, container } = render(<NewMovie />);
  expect(getByTestId("page-title").textContent).toBe("New Movie");
  expect(container.firstChild).toMatchSnapshot();
});
