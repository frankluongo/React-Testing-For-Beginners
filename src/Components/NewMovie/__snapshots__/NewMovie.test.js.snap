// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewMovie /> 1`] = `
<section>
  <h2
    data-testid="page-title"
  >
    New Movie
  </h2>
  <form
    data-testid="movie-form"
  >
    <label
      for="movie-text"
    >
      Movie
    </label>
    <input
      data-testid="movie-text-input"
      type="text"
      name="movie"
      id="movie-text"
      value=""
    />
    <button
      data-testid="movie-form-submit"
      type="submit"
    >
      Submit
    </button>
  </form>
</section>
`;
